username_api.py

```diff
@@ -19,6 +25,42 @@
 # Discussion: github.com/manu-chroma/username-availability-checker/issues/79
 session = r.Session()
 
+logger = logging.getLogger(__name__)
+handler = logging.StreamHandler(sys.stdout)
+formatter = logging.Formatter('[%(levelname)s|%(filename)s:%(lineno)s] '
+                              '%(asctime)s %(message)s')
+handler.setLevel(logging.WARNING)
+handler.setFormatter(formatter)
+logger.addHandler(handler)
+
+DEBUG = False
+
+
+def log_response(logger, msg, response, *args, **kwargs):
+    """
+        log_response saves response text under /static/errors/
+        with random file name.
+        This assumes that response object has text property
+        which is the content of response.
+    """
+    # Generates unique id for this response.
+    id = uuid.uuid4().hex
+
+    filename = 'static/errors/{}.dump'.format(id)
+
+    os.makedirs(os.path.dirname(filename), exist_ok=True)
+
+    with open(filename, 'w') as f:
+        # Note: https://toolbelt.readthedocs.io/en/latest/dumputils.html
+        f.write(dump.dump_all(response).decode('utf-8'))
+
+    log_if_DEBUG(logging.WARNING, msg, *args, **kwargs)
+    log_if_DEBUG(logging.WARNING, 'id: {}'.format(id), *args, **kwargs)
+
+
+def log_if_DEBUG(level, msg, *args, **kwargs):
+    if DEBUG: logging.log(level, msg, *args, **kwargs)
```

Comment:
why not use ``logging.DEBUG`` directly? 

Pls refer: https://docs.python.org/3/library/logging.html#logging.debug

Reply:
afaik, `logging.DEBUG` is a logging level.
also, note that pytest only captures log that is higher than `logging.INFO`.(in default config)

Reply:
https://docs.python.org/3/library/logging.html#logging.warning

Reply:
ah, you were talking about using `logging.warning` instead of `logging.log`.
well, i thought making a wrapper only for a level is not so good.

Reply:
I meant check that section out. In particular, I was referring to https://docs.python.org/3/library/logging.html#logging.debug because you were trying to debug log. 

Once you set the logging level in top, you can directly use these logging levels inside the code